community/values/src/main/java/org/neo4j/values/VirtualValue.java

```diff
@@ -29,15 +29,28 @@
  */
 public abstract class VirtualValue extends AnyValue
 {
-
     @Override
     public final boolean equals( Object other )
     {
-        if ( (other instanceof SequenceValue) && (this instanceof SequenceValue) )
+        if ( other == null )
+        {
+            return false;
+        }
+
+        // suppose that we will only ever call equals with Objects of AnyValue but just in case
+        try
+        {
+            if ( ((AnyValue) other).isSequenceValue() && this.isSequenceValue() )
```

Comment:
doing
```
try
{
(AnyValue) other)
}
catch ( ClassCastException e )
{
return false
}
```
is much more slower and not that easy to read compared to using instanceOf operator, so would suggest here to use instanceOf AnyValue instead. 
 This also applies to ArrayValue `equals`